From: Teo Mrnjavac <teo@getamarok.com>
Subject: [PATCH] No default playlist layout on start
 Unfortunately we have just detected a bug in the 2.2.1 release. This
 bug only affects a minority of users, but we decided it's safer to fix
 it anyway. Symptoms of this bug are:
 .
 * If Amarok is started without any existing configuration files, the
 playlist layout may appear corrupted. *
Origin: upstream, https://bugs.kde.org/show_bug.cgi?id=211717#c11
Bug: https://bugs.kde.org/show_bug.cgi?id=211717

---
 src/playlist/layouts/LayoutManager.cpp            |    2 ++
 src/playlist/layouts/PlaylistLayoutEditDialog.cpp |    5 ++++-
 2 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/src/playlist/layouts/LayoutManager.cpp b/src/playlist/layouts/LayoutManager.cpp
index b2903ee..541f989 100644
--- a/src/playlist/layouts/LayoutManager.cpp
+++ b/src/playlist/layouts/LayoutManager.cpp
@@ -55,6 +55,8 @@ LayoutManager::LayoutManager()
 
     KConfigGroup config = Amarok::config("Playlist Layout");
     m_activeLayout = config.readEntry( "CurrentLayout", "Default" );
+    if( !layouts().contains( m_activeLayout ) )
+        m_activeLayout = "Default";
     Playlist::ModelStack::instance()->top()->setGroupingCategory( activeLayout().groupBy() );
 }
 
diff --git a/src/playlist/layouts/PlaylistLayoutEditDialog.cpp b/src/playlist/layouts/PlaylistLayoutEditDialog.cpp
index 6d65590..9a061f2 100644
--- a/src/playlist/layouts/PlaylistLayoutEditDialog.cpp
+++ b/src/playlist/layouts/PlaylistLayoutEditDialog.cpp
@@ -410,7 +410,10 @@ void PlaylistLayoutEditDialog::reject()     //SLOT
     DEBUG_BLOCK
 
     debug() << "Applying initial layout: " << m_firstActiveLayout;
-    LayoutManager::instance()->setActiveLayout( m_firstActiveLayout );
+    if( layoutListWidget->findItems( m_firstActiveLayout, Qt::MatchExactly ).isEmpty() )
+        LayoutManager::instance()->setActiveLayout( "Default" );
+    else
+        LayoutManager::instance()->setActiveLayout( m_firstActiveLayout );
 
     QDialog::reject();
 }
-- 
tg: (aa942d4..) general/playlist_default_layout_fix (depends on: upstream)
